### ISSUE TITLE
[Accessibility]: Tags page: Search input has implicit label

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#35468]
### Audit type

Screen reader navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34425

### Problem description

<img width="1196" alt="image" src="https://user-images.githubusercontent.com/9516420/168589205-2e4cb07b-3d06-4324-ae37-45ac5cd7ce3f.png">

### Expected behavior

We should give this a dedicated label (`aria-label="Search tags"`)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.
